api/BUILD

```diff
@@ -61,3 +64,18 @@ bool_flag(
     build_setting_default = False,
     deprecation = "The value of this flag is ignored. Bazel builds always depend on Abseil for its pre-adopted `std::` types. You should remove this flag from your build command.",
 )
+
+int_flag(
+    name = "abi_version_no",
+    build_setting_default = 1,
```

Comment:
this is fine as is IMO but you can also add `make_variable = "ABI_VERSION_NO`, then add `toolchains = [":abi_version_no"],` to the rule above then you can use it like `defines = ["OPENTELEMETRY_ABI_VERSION_NO=$(ABI_VERSION_NO)"],` which allows you to remove the select

Reply:
Thanks for looking at it.

Actually, I prefer the current flags then, because it matches what is documented for CMake:

```
#
# We do not want to have WITH_ABI_VERSION = "1" or "2", and instead prefer two
# distinct flags, WITH_ABI_VERSION_1 and WITH_ABI_VERSION_2.
#
# This allows:
#
# * to have a specific option description for each ABI
# * to mark experimental/stable/deprecated on flags, for clarity
# * to search for exact abi usage move easily, discouraging:
#
#   * cmake -DWITH_ABI_VERSION=${ARG}
#
# While not supported, having distinct WITH_ABI_VERSION_1 and WITH_ABI_VERSION_2
# flags also opens the possibility to support multiple ABI concurrently, should
# that become necessary.
#
```

Reply:
note that the UX wouldn't change, only how it was consumed in the rule above, but yea sounds fine to me